spec/features/users/login_spec.rb

```diff
@@ -62,6 +60,86 @@
       expect(page).to have_content("Welcome Admin, #{@admin.name}!")
     end
 
+    it "cannot login user with incorrect email" do
+      expect(current_path).to eq("/login")
+
+      fill_in :email, with: "incorrect email"
+      fill_in  :password, with: @user.password
+      click_button "Login"
+
+      expect(current_path).to eq("/login")
+
+      expect(page).to have_content("Incorrect email or password.")
+    end
+
+    it "cannot login user with incorrect password" do
+      expect(current_path).to eq("/login")
+
+      fill_in :email, with: @user.email
+      fill_in  :password, with: "incorrect password"
+      click_button "Login"
+
+      expect(current_path).to eq("/login")
+
+      expect(page).to have_content("Incorrect email or password.")
+    end
+
+    it "cannot login user with incorrect email and incorrect password" do
+      expect(current_path).to eq("/login")
+
+      fill_in :email, with: "incorrect email"
+      fill_in  :password, with: "incorrect password"
+      click_button "Login"
+
+      expect(current_path).to eq("/login")
+
+      expect(page).to have_content("Incorrect email or password.")
+    end
+
+    it "can redirect a logged in regular user to profile" do
+      fill_in :email, with: @user.email
+      fill_in  :password, with: @user.password
+      click_button "Login"
+
+      visit "/merchants"
+      visit "/login"
+      expect(current_path).to eq("/profile")
+    end
+
+    it "can redirect a logged in merchant user to dashboard" do
+      fill_in :email, with: @merchant.email
+      fill_in  :password, with: @merchant.password
+      click_button "Login"
+
+      visit "/merchants"
+      visit "/login"
+      expect(current_path).to eq("/merchants/dashboard")
+    end
+
+    it "can redirect a logged in admin user to dashboard" do
+      fill_in :email, with: @admin.email
+      fill_in  :password, with: @admin.password
+      click_button "Login"
+
+      visit "/merchants"
+      visit "/login"
+      expect(current_path).to eq("/admin/dashboard")
+    end
+
+    it "cannot redirect a logged in regular user to a dashboard" do
```

Comment:
I don't think this is a very good sad path test, but I really just wanted to do a ```save_and_open_page``` and see what kind of redirect is occurring - this might be why our render isn't working - it already has an error page!